src/jquery.mustache.js

```diff
@@ -127,17 +127,23 @@
 	 * Loads the external Mustache templates located at the supplied URL and registers them for later use.  This method
 	 * returns a jQuery Promise and also support an `onComplete` callback.
 	 *
-	 * @param url			URL of the external Mustache template file to load.
+	 * @param url|settings  URL of the external Mustache template file to load, or a set of key/value pairs that 
+	 *						configure the Ajax request. All settings are optional except url. DataType setting will be
+	 *						overwrite.
 	 * @param onComplete	Optional callback function which will be invoked when the templates from the supplied URL
 	 *						have been loaded and are ready for use.
 	 * @returns				jQuery deferred promise which will complete when the templates have been loaded and are
 	 *						ready for use.
 	 */
 	function load(url, onComplete) {
-		return $.ajax({
-				url: url,
-				dataType: options.externalTemplateDataType
-			}).done(function (templates) {
+
+		var obj = {};
+
+		obj = typeof url === 'object' ? obj = url: obj.url = url;
```

Comment:
Thanks!  Could you fix up the white space on your last commit and throw an
Error instance instead of a string, please.  Also, could you squash your
commits into one and I'll merge it through tonight.

Thanks again
Jonny.

On 21 July 2015 at 17:19, Roger Russel <notifications@github.com> wrote:

> In src/jquery.mustache.js
> <https://github.com/jonnyreeves/jquery-Mustache/pull/38#discussion_r35121956>
> :
>
> >  	 * @param onComplete	Optional callback function which will be invoked when the templates from the supplied URL
> >  	 *						have been loaded and are ready for use.
> >  	 * @returns				jQuery deferred promise which will complete when the templates have been loaded and are
> >  	 *						ready for use.
> >  	 */
> >  	function load(url, onComplete) {
> > -		return $.ajax({
> > -				url: url,
> > -				dataType: options.externalTemplateDataType
> > -			}).done(function (templates) {
> > +
> > +		var obj = {};
> > +
> > +		obj = typeof url === 'object' ? obj = url: obj.url = url;
>
> Done, if url is null it will throw an exeception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonnyreeves/jquery-Mustache/pull/38/files#r35121956>.
>